src/game.py

```diff
@@ -90,13 +94,29 @@ def handle_move(location, die_roll):
                     print('%s has won!' % self.board.who_won())
                 break
 
-    def make_move(self, dice_roll, colour):
-        for die_roll in dice_roll:
-            valid_pieces = self.board.get_pieces(colour)
-            for piece in valid_pieces:
-                if self.board.is_move_possible(piece, die_roll):
-                    self.board.move_piece(piece, die_roll)
-                    break
+    def get_rolls_to_move(self, location, requested_move, available_rolls):
+        if available_rolls.__contains__(requested_move):
+            if self.board.is_move_possible(self.board.get_piece_at(location), requested_move):
+                return [requested_move]
+            return None
+        if len(available_rolls) == 1:
+            return None
```

Comment:
The first bit of this function isn't really needed. However I'm optimising for the case where the requested_move is one of the rolls (and avoiding the need to copy the board). I'm doing this to keep experiments as fast as possible